### PR TITLE
IServerAddressesFeature, logging

### PR DESCRIPTION
**Pull request by Tratcher**
#331, #328
@lodejard @davidfowl @DamianEdwards @halter73 @muratg 
I'll link the related PRs for Kestrel and WebListener shortly.

### CODE REVIEW

**Comment by davidfowl** [comment]
- Expose the info to Program.Main and Console.WriteLine from in there.

**Comment by Tratcher** [comment]
Updated.

**Comment by Tratcher** [comment]
```
Hosting environment: Production
Now listening on: http://+:5000/
Application started. Press Ctrl+C to shut down.
```
Also https://github.com/aspnet/Hosting/issues/351

**Comment by davidfowl** [comment]
:shipit:

File: src/Microsoft.AspNet.Hosting/Program.cs [hunk 1]
```diff
@@ -22,7 +24,7 @@ public Program(IServiceProvider serviceProvider)
 
         public void Main(string[] args)
         {
-            // Allow the location of the ini file to be specified via a --config command line arg
+            // Allow the location of the json file to be specified via a --config command line arg
```
Comment (halter73) [inline src/Microsoft.AspNet.Hosting/Program.cs]: Are ini files no longer supported?

Comment (Tratcher) [inline src/Microsoft.AspNet.Hosting/Program.cs]: correct, we switched to json recently.

**Pull request merged**